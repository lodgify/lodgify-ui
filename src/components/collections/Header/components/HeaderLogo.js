import React from 'react';
import { string, oneOf } from 'prop-types';
import { Menu, Image } from 'semantic-ui-react';
import classnames from 'classnames';

import { Heading } from 'typography/Heading';
import { Paragraph } from 'typography/Paragraph';
import { FlexContainer } from 'layout/FlexContainer';

export const HeaderLogo = ({
  logoHref,
  logoSubText,
  logoText,
  logoSrc,
  logoSize,
  logoSizes,
  logoSrcSet,
}) => (
  <Menu.Item
    className={classnames({
      'has-sub-text': !!logoSubText,
    })}
    href={logoHref}
    link
  >
    <FlexContainer flexDirection="column" justifyContent="center">
      {logoSrc ? (
        <Image
          alt={logoText}
          className={`${logoSize}-logo`}
          sizes={logoSizes}
          src={logoSrc}
          srcSet={logoSrcSet}
        />
      ) : (
        <Heading className="header-title" size="small">
          {logoText}
        </Heading>
      )}
      {logoSubText && <Paragraph size="tiny">{logoSubText}</Paragraph>}
    </FlexContainer>
  </Menu.Item>
);

HeaderLogo.displayName = 'HeaderLogo';

HeaderLogo.defaultProps = {
  logoHref: '',
  logoSubText: '',
  logoSize: 'medium',
  logoSrc: undefined,
  logoSizes: undefined,
  logoText: '',
  logoSrcSet: undefined,
};

HeaderLogo.propTypes = {
  logoHref: string,
  logoSize: oneOf(['medium', 'large', 'huge']),
  logoSizes: string,
  logoSrc: string,
  logoSrcSet: string,
  logoSubText: string,
  logoText: string,
};
