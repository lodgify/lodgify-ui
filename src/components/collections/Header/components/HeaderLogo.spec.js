import React from 'react';
import { mount } from 'enzyme';

import { HeaderLogo } from './HeaderLogo';

const mountHeaderLogo = props => mount(<HeaderLogo {...props} />);

describe('HeaderLogo', () => {
  describe('passing `logoSubText`', () => {
    it('should return a paragraph with the sub text', () => {
      const header = mountHeaderLogo({ logoSubText: 'foo' });

      expect(header).toMatchSnapshot();
    });
  });
  describe('passing `logoSrc`', () => {
    it('should return an image with the sub text', () => {
      const header = mountHeaderLogo({ logoSrc: 'foo' });

      expect(header).toMatchSnapshot();
    });
  });
});
