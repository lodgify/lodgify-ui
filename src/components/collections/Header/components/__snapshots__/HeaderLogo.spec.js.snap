// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo passing \`logoSrc\` should return an image with the sub text 1`] = `
<HeaderLogo
  logoHref=""
  logoSize="medium"
  logoSrc="foo"
  logoSubText=""
  logoText=""
>
  <MenuItem
    className=""
    href=""
    link={true}
  >
    <div
      className="link item"
      href=""
      onClick={[Function]}
    >
      <FlexContainer
        alignContent={null}
        alignItems={null}
        className={null}
        flexDirection="column"
        flexWrap={null}
        justifyContent="center"
      >
        <div
          className="flex-container"
          style={
            Object {
              "alignContent": null,
              "alignItems": null,
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": "1",
              "flexWrap": null,
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <Image
            alt=""
            as="img"
            className="medium-logo"
            src="foo"
            ui={true}
          >
            <img
              alt=""
              className="ui image medium-logo"
              src="foo"
            />
          </Image>
        </div>
      </FlexContainer>
    </div>
  </MenuItem>
</HeaderLogo>
`;

exports[`HeaderLogo passing \`logoSubText\` should return a paragraph with the sub text 1`] = `
<HeaderLogo
  logoHref=""
  logoSize="medium"
  logoSubText="foo"
  logoText=""
>
  <MenuItem
    className="has-sub-text"
    href=""
    link={true}
  >
    <div
      className="link item has-sub-text"
      href=""
      onClick={[Function]}
    >
      <FlexContainer
        alignContent={null}
        alignItems={null}
        className={null}
        flexDirection="column"
        flexWrap={null}
        justifyContent="center"
      >
        <div
          className="flex-container"
          style={
            Object {
              "alignContent": null,
              "alignItems": null,
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": "1",
              "flexWrap": null,
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <Heading
            className="header-title"
            hasMargin={true}
            isColorInverted={false}
            size="small"
            textAlign={null}
          >
            <Header
              as="h4"
              className="header-title"
              inverted={false}
              textAlign={null}
            >
              <h4
                className="ui header header-title"
              />
            </Header>
          </Heading>
          <Paragraph
            isCompact={false}
            size="tiny"
            title="foo"
            weight={null}
          >
            <p
              className="tiny"
            >
              foo
            </p>
          </Paragraph>
        </div>
      </FlexContainer>
    </div>
  </MenuItem>
</HeaderLogo>
`;
